### ISSUE TITLE
Isn't compatible with new YouTube design

### ISSUE THREAD

**Issue by RunninglVlan** [demalexx/youtube-opera-speeddial#3]
Hi, could you please update extension to make it work with new YouTube design?
For me it fails in updateData().loadCallback(event) - `executeSelector(response_container, 'is_items_list')` returns false and `executeSelector(response_container, 'items_grid')` throws following error:
`Uncaught Selector can't find items_grid (.yt-lockup-video)`

**Comment by demalexx** [comment]
Hi. You use grid layout on YouTube? I use list layout and it works fine except it doesn't show watched videos (they replaced watched badge by progress indicator on the bottom).

**Comment by RunninglVlan** [comment]
Are you sure you're using new design and haven't switched back to old one? Switching layouts doesn't help me. I get the same error.

**Comment by demalexx** [comment]
I don't see any change to layout and I can't find where I could turn it on. Probably I'll get this update later.

**Comment by RunninglVlan** [comment]
Here's a teaser for you:
New YT Grid
![New YT Grid](http://i.imgur.com/AsriDG4.jpg)
New YT List
![New YT List](http://i.imgur.com/CYWQ3rm.png)
;)

**Comment by demalexx** [comment]
Thanks :) Yeah it looks a bit different (and seems they changed internal html again 😒). I'll update extension once I get new style. Sorry that extension won't work until then.

**Comment by RunninglVlan** [comment]
BTW, what I like about new design is that they finally added option to add videos to playlists from the home page. This options was available for a long time in their mobile application.

**Comment by RunninglVlan** [comment]
Did you get new style yet? BTW, can you use your https://github.com/demalexx/youtube-opera-speeddial/blob/master/selectors.json to fix this issue?

**Comment by demalexx** [comment]
Yep, still old design. May be you turned on some experimental feature or installed extension that updates YouTube pages?

Yeah probably I could use selectors.json but that wouldn't best solution.

Could you please send me html of your version of YouTube Subscriptions page in list and grid layouts? I'll try to think something so extension could work ok for people with different layouts.

**Comment by RunninglVlan** [comment]
Here you go
[Subscriptions.zip](https://github.com/demalexx/youtube-opera-speeddial/files/718927/Subscriptions.zip)

**Comment by RunninglVlan** [comment]
I did small analysis of this issue (tried to update `SELECTORS`) and I'm afraid that your extension won't work with YT new design at least in its current state. It seems there is no content of videos from subscriptions in source when you get data from `XMLHttpRequest`. Looks like it's dynamically loaded afterwards. But I hope you will still fix this interesting issue. ;) Good luck.
In the meantime I'll try using [this approach](https://support.google.com/youtube/answer/6224202?hl=en) through [The Old Reader](https://addons.opera.com/en/extensions/details/the-old-reader-extension/?display=en).

**Comment by demalexx** [comment]
Thanks.

Are you sure content is loaded dynamically? This sounds very bad, because I thought about other solutions and they don't seem good.

1. I wanted to use ugly YouTube API, but, in short, it's either would update videos very slow (due to a lot of requests), or would exhaust API calls quota very quickly.

2. You gave me idea with using RSS. But it's again a lot of queries (for each subscription), and I'm afraid, YouTube keeps that RSS just because they forget about it, and one day, when they found this working RSS, they turn it off.

So even if I don't like selectors approach very much, it seems the best option available. If content is loaded dynamically it's very bad. Need to use one of worse options. 

IIRC YouTube does return one html and then updates it by js, but that's ok - html from the server could be parsed.
